cookbooks/bach_krb5/recipes/keytabs.rb

```diff
@@ -59,5 +104,24 @@
   end
 end
 
+if node[:bcpc][:hadoop][:kerberos][:additional][:hosts].count > 0
+  # Generate keytabs for additional hosts
+  node[:bcpc][:hadoop][:kerberos][:additional][:hosts].each do |h|
+    node[:bcpc][:hadoop][:kerberos][:data].each do |ke, va|
+
+      service_principal = va['principal']
+      host_fqdn = h
+      keytab_file = "#{h.split(".")[0]}_#{va['principal']}"
+
+      # Create the keytab file
+      execute "creating-keytab-for-#{ke}" do
+        command "kadmin.local -q 'xst -k #{keytab_dir}/#{host_fqdn}/#{keytab_file} -norandkey #{service_principal}/#{host_fqdn}@#{realm} HTTP/#{host_fqdn}@#{realm}'"
```

Comment:
if a keytab exists the block will not run, however if it does not do you even care if the KVNO is incremented?